### PR TITLE
exclude legacy types from rate limiting

### PR DESCRIPTION
**Pull request by bergarces**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Preemptively adds to the rate limiting exclusion list several features that are currently not triggering the confirmation by the approval controller, but that will in the future (based on the confirmations system team planning).

This PR should not cause any feature change.

**Screenshots/Recordings**

[personal_sign.webm](https://github.com/MetaMask/metamask-mobile/assets/1970725/323520bf-9cfd-4f96-bc9e-30f887b8e5f5)

**Issue**

Progresses [#???](https://github.com/MetaMask/mobile-planning/issues/979)

**Checklist**

* [X] There is a related GitHub issue
* [ ] Tests are included if applicable
* [X] Any added code is fully documented

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -114,7 +114,7 @@
     "react-native-level-fs/**/semver": "^4.3.2",
     "@metamask/contract-metadata": "^2.1.0",
     "@metamask/controller-utils": "~3.0.0",
-    "@metamask/approval-controller": "~2.0.0",
+    "@metamask/approval-controller": "2.1.0",
```
Comment (bergarces) [inline package.json:117]: So... this is the minimum version required to have access to the exclusion list.

It all seems to build and work fine with it.

Comment (matthewwalsh0) [inline package.json:117]: Shall we go with `^2.0.0` as we try and be strict with the core versioning?

Comment (matthewwalsh0) [reply package.json:117]: As discussed in the DSM, better to be safe for now.

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by bergarces** [comment]
@SocketSecurity ignore-all

**Review by vinistevam — APPROVED** [review]

**Pull request merged**